### PR TITLE
feat: setup DynamoDbMapper publication 

### PR DESCRIPTION
**Pull request by lauzadis**
This PR sets up DynamoDBMapper and other related libraries for publication. It also removes the dynamodb-mapper-annotation-processor-test module since it's [now duplicated by SchemaGeneratorPluginTest](https://github.com/awslabs/aws-sdk-kotlin/blob/feat-ddb-mapper/hll/dynamodb-mapper/dynamodb-mapper-schema-generator-plugin/src/test/kotlin/aws/sdk/kotlin/hll/dynamodbmapper/plugins/SchemaGeneratorPluginTest.kt#L336) 

## Description of changes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: hll/build.gradle.kts [hunk 1]
```diff
@@ -31,22 +33,32 @@ val optinAnnotations = listOf(
...
 )
 
+@OptIn(InternalApi::class)
+fun getHllPreviewVersion() = if (sdkVersion.contains("-SNAPSHOT")) { // i.e. 1.3.29-beta-SNAPSHOT
+    sdkVersion
+        .removeSuffix("-SNAPSHOT")
+        .ensureSuffix("-beta-SNAPSHOT")
+} else {
+    sdkVersion.ensureSuffix("-beta") // i.e. 1.3.29-beta
+}
```
Comment (ianbotsf) [inline hll/build.gradle.kts:43]: **Nit**: "[i.e.](https://en.wiktionary.org/wiki/i.e.)" → "[e.g.](https://en.wiktionary.org/wiki/e.g.)"

Comment (ianbotsf) [inline hll/build.gradle.kts:43]: **Question**: Any reason not to make this a `val` instead of a `fun`?

File: hll/hll-codegen/build.gradle.kts [hunk 2]
```diff
@@ -37,10 +37,6 @@ dependencies {
     testImplementation(libs.kotlin.test.junit5)
 }
 
-val sdkVersion: String by project
-group = "aws.sdk.kotlin"
-version = sdkVersion
-
```
Comment (ianbotsf) [inline hll/hll-codegen/build.gradle.kts:43]: **Question**: What version will **hll-codegen** use now?

File: hll/dynamodb-mapper/dynamodb-mapper-schema-generator-plugin/build.gradle.kts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import aws.smithy.kotlin.runtime.InternalApi
+import aws.smithy.kotlin.runtime.text.ensureSuffix
```
Comment (ianbotsf) [inline hll/dynamodb-mapper/dynamodb-mapper-schema-generator-plugin/build.gradle.kts:2]: **Question**: How are these imports resolvable by the build script? Do we have the SDK/Smithy runtime in the classpath at this point?

Comment (lauzadis) [reply hll/dynamodb-mapper/dynamodb-mapper-schema-generator-plugin/build.gradle.kts:2]: Not sure _why_ it works like this, but we do have a dependency on **smithy-kotlin:runtime-core** in this module, and the imports to resolve

File: hll/hll-codegen/build.gradle.kts (hunk 2 above)
Comment (lauzadis) [reply hll/hll-codegen/build.gradle.kts:43]: It gets the version from the parent module **hll**: https://github.com/awslabs/aws-sdk-kotlin/pull/1419/files/fdf75ed58ff28d75e36f8d52662480a0c3d72ff1#diff-bc29db26a69fa060c49b8d3399f5ee9eb8014006435f213e5f424c0b0656be7eR47

File: hll/build.gradle.kts [hunk 4]
```diff
@@ -31,22 +33,32 @@ val optinAnnotations = listOf(
...
+subprojects {
+    group = "aws.sdk.kotlin"
+    version = getHllPreviewVersion()
+    configurePublishing("aws-sdk-kotlin")
+}
+
 subprojects {
     if (!needsKmpConfigured) {
         return@subprojects
     }
```
[the remark this answers is not in the document]
Comment (lauzadis) [reply hll/build.gradle.kts:54]: All subprojects should have those values configured, they are not KMP-specific which is what the first block is for

File: hll/hll-codegen/build.gradle.kts (hunk 2 above)
Comment (ianbotsf) [reply hll/hll-codegen/build.gradle.kts:43]: OK so we'll just re-add `version = sdkVersion`  when it comes time to GA the mapper?

**Review by ianbotsf — APPROVED** [review]

**Review by 0marperez — APPROVED** [review]

File: hll/hll-codegen/build.gradle.kts (hunk 2 above)
Comment (lauzadis) [reply hll/hll-codegen/build.gradle.kts:43]: Yes and it's set up so we just have to change it in that one spot

File: hll/dynamodb-mapper/dynamodb-mapper-schema-generator-plugin/build.gradle.kts [hunk 5]
```diff
@@ -69,6 +88,17 @@ tasks.test {
...
+
+@OptIn(InternalApi::class)
+fun getHllPreviewVersion() = if (sdkVersion.contains("-SNAPSHOT")) { // i.e. 1.3.29-beta-SNAPSHOT
+    sdkVersion
+        .removeSuffix("-SNAPSHOT")
+        .ensureSuffix("-beta-SNAPSHOT")
+} else {
+    sdkVersion.ensureSuffix("-beta") // i.e. 1.3.29-beta
+}
+
```
[the remark this answers is not in the document]
Comment (lauzadis) [reply hll/dynamodb-mapper/dynamodb-mapper-schema-generator-plugin/build.gradle.kts:101]: Personally I don't think this function will last long enough, it should be removed once we GA the DynamoDbMapper

**Pull request merged**